doc/user_mgmt/user_mgmt_hld.md

```diff
@@ -0,0 +1,595 @@
+
+In addition, we will define two default users:
+* **admin**: User role is admin.<br/>
+	The only change from the current admin in Sonic is the additional group: adm
+
+* **monitor**: User role is monitor. <br/>
+	User with monitoring permissions only .<br/>
+
+**Note**:
+ Group adm is used for system monitoring tasks. Members of this group can read the files under "/var/log"
+
+
+###  2.3 <a name='UsercfdRole'></a>Usercfgd Role:
+
+Usercfgd is the daemon that get events from USER_TABLE in CONF_DB and apply the new configuration on the system.<br/>
+The new daemon should start running after "config-setup.service" (i.e. after database is ready) but before systemd enables "system-logind.service" (i.e. before user can login to system)<br/>
+
+
+![services_diagram](services_diagram.png))<br/>
+
+
+##  3 <a name='Configurationandmanagement'></a>Configuration and management
+
+###  3.1 <a name='ConfigDBTables'></a>ConfigDB Tables
+
+```
+USER_TABLE:{
+  "<username>":{
+    "state": {{enabled/disabled}}
+    "full-name": {{string}}
+    "hashed-password": {{string}}
+    "hashed-password_history": {{string}}
+    "role": {{admin/monitor}}
+    "passwd_date_of_last_change": {{uint32}}
+  }
+}
+```
+
+```
+ROLE_TABLE:{
+  "<role-name>":{
+    "primary_group": {{string}}
+    "secondary_groups": {{string}}
+  }
+}
+```
+###  3.2 <a name='ConfigDBschemas'></a>ConfigDB schemas
+
+```
+; Defines schema for User configuration attributes in USER_TABLE:
+key                                   = "<username>"                ;user configuration
+; field                               = value
+STATE                                 = "enabled" / "disabled"      ; user enable/disable
+FULL-NAME                             = STRING                      ; Full-name/Description of user
+HASHED-PASSWORD                       = STRING                      ; Hashed password
+HASHED-PASSWORD_HISTORY               = STRING                      ; List of old passwords (0-99)
+PASSWD_DATE_OF_LAST_CHANGE            = UINT32                      ; date of password last change
+ROLE                                  = "admin" / "monitor"         ; Role/Capability
+
+```
+
+```
+; Defines schema for role configuration attributes in ROLE_TABLE:
+key                                   = "<role name>"               ;Role configuration
+; field                               = value
+PRIMARY_GROUP                         = STRING                      ; Primary linux group
+SECONDARY_GROUP                       = STRING                      ; List of secondary linux groups
+```
+
+###  3.3 <a name='CLIYANGmodel'></a>CLI/YANG model
+
+##### YANG model
+```yang
+
+module sonic-user-mgmt {
+    yang-version 1.1;
+    namespace "http://github.com/Azure/sonic-user-mgmt";
+	prefix user-mgmt;
+
+    description "User Management YANG Module for SONiC OS";
+
+	revision 2022-12-01 {
+        description "First Revision";
+    }
+   container sonic-user-mgmt {
+
+        container USER_TABLE {
```

Comment:
Can we have the container name as "USER" instead of USER_TABLE? generally..we use _TABLE suffix in the APP DB tables.

Reply:
I have seen examples in CONFIG DB for tables with "_TABLE" Suffix: ACL_TABLE, FLEX_COUNTER_TABLE, PBH_TABLE...
Is there a guideline document ?

Reply:
I understand few exceptions today, IMO we should not keep that has a practice for new tables.

Reply:
https://github.com/sonic-net/SONiC/blob/master/doc/mgmt/SONiC_YANG_Model_Guidelines.md